q2_demux/_demux.py

```diff
@@ -251,7 +252,17 @@ def _make_barcode_map(barcodes, rev_comp_mapping_barcodes):
                              '%d != %d. Variable length barcodes are not '
                              'supported.' % (len(barcode), barcode_len))
         if rev_comp_mapping_barcodes:
-            barcode = str(skbio.DNA(barcode).reverse_complement())
+            try:
+                barcode = str(skbio.DNA(barcode).reverse_complement())
+            except ValueError as ve:
+                if re.match('^ValueError[(]"Invalid characters in sequence[.,'
+                            ' \n]*',
```

Comment:
Nice regex, you can also spell `(` as `\(` to prevent it from being an unmatched capture group, I don't think I've seen this style `[(]` before but it makes sense. You might have run into issues with `\` itself needing to be escaped by python. This is where the `r""` strings can be handy.


All that said, matching on `Invalid characters in sequence` on `str(ve)` would probably be enough.

Reply:
(No changes needed here, btw)

Reply:
That is exactly the issue that we ran into 😂